src/coreml.cs

```diff
@@ -0,0 +1,269 @@
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLFeatureValue : NSCopying {
+
+		[Export ("type")]
+		MLFeatureType Type { get; }
+
+		[Export ("undefined")]
+		bool Undefined { [Bind ("isUndefined")] get; }
+
+		[Export ("int64Value")]
+		long Int64Value { get; }
+
+		[Export ("doubleValue")]
+		double DoubleValue { get; }
+
+		[Export ("stringValue")]
+		string StringValue { get; }
+
+		[NullAllowed, Export ("multiArrayValue")]
+		MLMultiArray MultiArrayValue { get; }
+
+		[Export ("dictionaryValue")]
+		NSDictionary<NSObject, NSNumber> DictionaryValue { get; }
+
+#if !WATCH
+		[NullAllowed, Export ("imageBufferValue")]
+		CVPixelBuffer ImageBufferValue { get; }
+
+		[Static]
+		[Export ("featureValueWithPixelBuffer:")]
+		MLFeatureValue FromPixelBuffer (CVPixelBuffer value);
+#endif
+		[Static]
+		[Export ("featureValueWithInt64:")]
+		MLFeatureValue FromInt64 (long value);
+
+		[Static]
+		[Export ("featureValueWithDouble:")]
+		MLFeatureValue FromDouble (double value);
+
+		[Static]
+		[Export ("featureValueWithString:")]
+		MLFeatureValue FromString (string value);
+
+		[Static]
+		[Export ("featureValueWithMultiArray:")]
+		MLFeatureValue FromMultiArray (MLMultiArray value);
+
+		[Static]
+		[Export ("undefinedFeatureValueWithType:")]
+		MLFeatureValue CreateUndefined (MLFeatureType type);
+
+		[Static]
+		[Export ("featureValueWithDictionary:error:")]
+		[return: NullAllowed]
+		MLFeatureValue FromDictionary (NSDictionary<NSObject, NSNumber> value, out NSError error);
+
+		[Export ("isEqualToFeatureValue:")]
+		bool IsEqual (MLFeatureValue value);
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLModel {
+
+		[Export ("modelDescription")]
+		MLModelDescription ModelDescription { get; }
+
+		[Static]
+		[Export ("modelWithContentsOfURL:error:")]
+		[return: NullAllowed]
+		MLModel FromUrl (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("predictionFromFeatures:error:")]
+		[return: NullAllowed]
+		IMLFeatureProvider GetPrediction (IMLFeatureProvider input, out NSError error);
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLModelDescription {
+
+		[Export ("inputDescriptionsByName")]
+		NSDictionary<NSString, MLFeatureDescription> InputDescriptionsByName { get; }
+
+		[Export ("outputDescriptionsByName")]
+		NSDictionary<NSString, MLFeatureDescription> OutputDescriptionsByName { get; }
+
+		[NullAllowed, Export ("predictedFeatureName")]
+		string PredictedFeatureName { get; }
+
+		[NullAllowed, Export ("predictedProbabilitiesName")]
+		string PredictedProbabilitiesName { get; }
+
+		[Export ("metadata")]
+		[Internal]
+		NSDictionary _Metadata { get; }
+
+		[Wrap ("_Metadata")]
+		MLModelMetadata Metadata { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Static]
```

Comment:
I think they are more discoverable than the indexers but if you want them internal I'll follow suite :)